src/java.base/share/classes/java/io/Reader.java

```diff
@@ -140,6 +140,208 @@ public void close() {
         };
     }
 
+    /**
+     * Returns a new {@code Reader} whose source is a {@link CharSequence}.
+     *
+     * <p> The returned reader's class provides an API compatible with
+     * {@link StringReader}, but with no guarantee of synchronization, and
+     * without limitation to {@code String} sources. This class is designed for
+     * use as a drop-in replacement for {@code StringReader} in places where the
+     * reader was being used by a single thread (as is generally the case).
+     * Where possible, it is recommended that this class be used in preference
+     * to {@code StringReader} as it will be faster under most implementations.
+     *
+     * <p> The returned stream is initially open. The stream is closed by
+     * calling the {@code close()} method. Subsequent calls to {@code close()}
+     * have no effect.
+     *
+     * <p> After the stream has been closed, the {@code read()},
+     * {@code read(char[])}, {@code read(char[], int, int)},
+     * {@code read(CharBuffer)}, {@code ready()}, {@code skip(long)}, and
+     * {@code transferTo()} methods all throw {@code IOException}.
+     *
+     * <p> The {@code markSupported()} method returns {@code true}.
+     *
+     * <p> The {@link #lock object} used to synchronize operations on the
+     * returned {@code Reader} is not specified.
+     *
+     * @param c {@code CharSequence} providing the character stream.
+     *
+     * @return a {@code Reader} which reads characters from {@code c}
+     *
+     * @since 24
+     */
+    public static Reader of(CharSequence c) {
```

Comment:
Should we give this factory a more specific name so we don't clash in the future? For example, if we add another factory `of(A a)` for interface `A`, then it would be confusing to have an instance of `interface C extends CharSequence, A` to be passed to `of`.

Reply:
I do not see what is "confusing" in that case, as the caller still gets what he intends: a reader for the passed source. I also wonder how likely that case actually is. Anyways, I may be biased as I proposed `of`.

@AlanBateman WDYT?

Reply:
You'll get different opinions on naming, personally I think Reader.of(cs) is very readable at use-sites. There are several other static methods named "of" added in other areas in recent years.

Reply:
So let's go with `Reader.of()` unless nobody has stronger feelings against it. 😃